.github/workflows/solrj-test-crave.yml

```diff
@@ -0,0 +1,41 @@
+name: SolrJ Tests
+
+on:
+  pull_request:
+    branches:
+      - 'main'
+    paths:
+      - '.github/workflows/solrj-test.yml'
+      - 'solr/solrj/**'
+
+jobs:
+  test:
+    name: Run SolrJ Tests
+
+    runs-on: ubuntu-latest
+
+    steps:
+    # Setup
+    - uses: actions/checkout@v2
+    - name: Set up JDK 11
+      uses: actions/setup-java@v2
+      with:
+        distribution: 'temurin'
+        java-version: 11
+        java-package: jdk
+    - name: Grant execute permission for gradlew
+      run: chmod +x gradlew
+    - uses: actions/cache@v2
+      with:
+        path: |
+          ~/.gradle/caches
+        key: ${{ runner.os }}-gradle-solrj-${{ hashFiles('versions.lock') }}
+        restore-keys: |
+          ${{ runner.os }}-gradle-solrj-
+          ${{ runner.os }}-gradle-
+    - name: Get the Crave binary
+      run: curl -s https://raw.githubusercontent.com/accupara/crave/master/get_crave.sh | bash -s --
+    - name: Initialize gradle settings
+      run: ./crave run -- ./gradlew localSettings
```

Comment:
Re first comment: Yes, the output from `localSettings` will be retained between subsequent `crave run`s.     
Re second comment: I've changed the worker count to match the core count on the remote build/test node so that it matches whatever the ephemeral node is configured to be.

In other words, when we use a 16 core machine, we'll use 16 workers. When we configure crave to use `n2d-standard-224` instances, `nproc` will return 224 and automatically tell gradlew to use the maximum number of cores.

We can consider a linear scale factor for the number of cores in a later change as we understand or tweak how well gradle performs on multiple cores.

Reply:
Does that return the number of REAL cpu cores, or the same count you would see in something like /proc/cpuinfo?    I would wager that most modern CPUs support hyperthreading, which makes most programs think my 24 core server has 48 cores.  I don't want to disable HT, because even though it's not a true doubling of the core count, it does help some workloads run faster.  But I wouldn't want the solr build to decide it can handle 48 workers.

Reply:
@elyograg `nproc` seems to return "the number of cores available to your process". Through empirical testing on my X1 carbon, this seems to mean "hyperthreaded cores".

Testing on the `n1-standard-96` set up that we have for Solr testing got me:
```
19:23:12 uv@uv-x1:~/src/github.com/java/solr/solr$ crave ssh -- nproc
96
```
So it looks like it is hyperthreaded cores for sure.

Reply:
Half the number of available processors reported by the JVM; the JVM counts hyper threads individually.  Since 2x threads per core is common, our build divided this by two to guess at the number of physical cores.  Then this is capped at 12.  Logic is in `./gradle/generation/local-settings.gradle`.  There are lots of concurrent threads in the JVM so I this this approach of using physical cores in spite of hyper threading makes sense.  On a machine with 96 threads on 48 cores (which I used last weekend playing around on Crave.io), 48 gradle workers worked well (and so did 24).